homeassistant/components/qnap_qsw/button.py

```diff
@@ -0,0 +1,75 @@
+"""Support for the QNAP QSW buttons."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from typing import Final
+
+from aioqsw.const import QSD_PRODUCT, QSD_SYSTEM_BOARD
+
+from homeassistant.components.button import (
+    ButtonDeviceClass,
+    ButtonEntity,
+    ButtonEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import QswEntity
+from .const import DOMAIN, QSW_REBOOT
+from .coordinator import QswUpdateCoordinator
+
+
+@dataclass
+class QswButtonEntityDescription(ButtonEntityDescription):
+    """A class that describes QNAP QSW button entities."""
+
+    press_action: Callable | None = None
```

Comment:
Can we make `Callable` more specific? Add in expected arguments and return types.
Also, isn't this a required field? (As in cannot be None?)

Reply:
Fixed in d718361c9f03bf423a966a1921e8385e788d3cd8